app/components/channel_drawer/channel_list.js

```diff
@@ -147,22 +150,146 @@ class ChannelList extends React.Component {
     handleClose = (channel) => {
         const {formatMessage} = this.props.intl;
         Alert.alert(
-            formatMessage({id: 'mobile.channel_list.alertTitleCloseDM', defaultMessage: 'Close Direct Message'}),
+            formatMessage({id: 'mobile.channel_list.alertTitleRemoveDM', defaultMessage: 'Remove Direct Message'}),
             formatMessage({
-                id: 'mobile.channel_list.alertMessageCloseDM',
-                defaultMessage: 'Are you sure you want to close the DM with {username}'
+                id: 'mobile.channel_list.alertMessageRemoveDM',
+                defaultMessage: 'Are you sure you want to remove the DM with {username} from the list?'
             }, {
                 username: channel.display_name
             }),
-            [
-                {text: formatMessage({id: 'mobile.channel_list.alertNo', defaultMessage: 'No'})},
-                {
-                    text: formatMessage({id: 'mobile.channel_list.alertYes', defaultMessage: 'Yes'}),
-                    onPress: () => this.props.handleCloseDM(channel)
-                }]
+            [{
+                text: formatMessage({id: 'mobile.channel_list.alertNo', defaultMessage: 'No'})
+            }, {
+                text: formatMessage({id: 'mobile.channel_list.alertYes', defaultMessage: 'Yes'}),
+                onPress: () => {
+                    this.setState({showOptions: false});
+                    this.props.handleCloseDM(channel);
+                }
+            }]
         );
     };
 
+    handleLeave = (channel, term) => {
+        const {formatMessage} = this.props.intl;
+        Alert.alert(
+            formatMessage({id: 'mobile.channel_list.alertTitleLeaveChannel', defaultMessage: 'Leave {term}'}, {term}),
+            formatMessage({
+                id: 'mobile.channel_list.alertMessageLeaveChannel',
+                defaultMessage: 'Are you sure you want to leave the {term} with {name}?'
+            }, {
+                term: term.toLowerCase(),
+                name: channel.display_name
+            }),
+            [{
+                text: formatMessage({id: 'mobile.channel_list.alertNo', defaultMessage: 'No'})
+            }, {
+                text: formatMessage({id: 'mobile.channel_list.alertYes', defaultMessage: 'Yes'}),
+                onPress: () => {
```

Comment:
If a user chooses to leave a channel that they currently have open, the channel stays open in the channel view after they leave. We should move the user to one of their other channels right?